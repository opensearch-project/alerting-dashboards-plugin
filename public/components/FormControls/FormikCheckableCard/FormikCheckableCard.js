/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiCheckableCard } from '@elastic/eui';

import FormikInputWrapper from '../FormikInputWrapper';
import FormikFormRow from '../FormikFormRow';

const FormikCheckableCard = ({
  name,
  formRow = false,
  fieldProps = {},
  rowProps = {},
  inputProps = {},
}) => (
  <FormikInputWrapper
    name={name}
    fieldProps={fieldProps}
    render={({ field, form }) =>
      formRow ? (
        <FormikFormRow name={name} form={form} rowProps={rowProps}>
          <FieldCheckableCard name={name} form={form} field={field} inputProps={inputProps} />
        </FormikFormRow>
      ) : (
        <FieldCheckableCard name={name} form={form} field={field} inputProps={inputProps} />
      )
    }
  />
);

const FieldCheckableCard = ({
  name,
  form,
  field,
  inputProps: { onChange, id, label, ...rest },
}) => (
  <EuiCheckableCard
    name={name}
    id={id}
    label={label}
    {...field}
    {...rest}
    onChange={(e) =>
      typeof onChange === 'function' ? onChange(e, field, form) : field.onChange(e)
    }
  />
);

FormikCheckableCard.propTypes = {
  name: PropTypes.string.isRequired,
  formRow: PropTypes.bool,
  fieldProps: PropTypes.object,
  rowProps: PropTypes.object,
  inputProps: PropTypes.object,
};

FieldCheckableCard.propTypes = {
  name: PropTypes.string.isRequired,
  form: PropTypes.object.isRequired,
  inputProps: PropTypes.object.isRequired,
  field: PropTypes.object.isRequired,
};

export default FormikCheckableCard;
