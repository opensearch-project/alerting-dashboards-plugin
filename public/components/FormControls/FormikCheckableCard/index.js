/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import FormikCheckableCard from './FormikCheckableCard';

export default FormikCheckableCard;
