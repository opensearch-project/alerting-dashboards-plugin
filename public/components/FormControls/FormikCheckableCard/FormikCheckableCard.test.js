/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';
import { Formik } from 'formik';

import FormikCheckableCard from './FormikCheckableCard';

describe('FormikCheckableCard', () => {
  test('render formRow', () => {
    const component = (
      <Formik>
        <FormikCheckableCard name="testing" formRow />
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('render', () => {
    const component = (
      <Formik>
        <FormikCheckableCard name="testing" />
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });
});
