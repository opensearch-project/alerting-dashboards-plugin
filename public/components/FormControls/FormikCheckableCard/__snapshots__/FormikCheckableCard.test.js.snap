// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikCheckableCard render 1`] = `
<div
  class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard"
>
  <div
    class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
  >
    <div
      class="euiRadio euiRadio--noLabel"
    >
      <input
        class="euiRadio__input"
        name="testing"
        type="radio"
      />
      <div
        class="euiRadio__circle"
      />
    </div>
  </div>
  <div
    class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
  >
    <label
      class="euiCheckableCard__label"
    />
  </div>
</div>
`;

exports[`FormikCheckableCard render formRow 1`] = `
<div
  class="euiFormRow"
  id="testing-form-row-row"
>
  <div
    class="euiFormRow__fieldWrapper"
  >
    <div
      class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard"
    >
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
      >
        <div
          class="euiRadio euiRadio--noLabel"
        >
          <input
            class="euiRadio__input"
            name="testing"
            type="radio"
          />
          <div
            class="euiRadio__circle"
          />
        </div>
      </div>
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
      >
        <label
          class="euiCheckableCard__label"
        />
      </div>
    </div>
  </div>
</div>
`;
